src/coreclr/nativeaot/Runtime/GCHelpers.cpp

```diff
@@ -471,6 +481,43 @@ EXTERN_C int64_t QCALLTYPE RhGetTotalAllocatedBytesPrecise()
     return allocated;
 }
 
+inline void FireAllocationSampled(GC_ALLOC_FLAGS flags, size_t size, size_t samplingBudgetOffset, Object* orObject)
+{
+    // Note: this code is duplicated from GCToCLREventSink::FireGCAllocationTick_V4
+    void* typeId = nullptr;
+    const WCHAR* name = nullptr;
+    // TODO: this does not compile due to duplicated symbols when sstring.h is included
+    //InlineSString<MAX_CLASSNAME_LENGTH> strTypeName;
+    //EX_TRY
+    //{
+    //    TypeHandle th = GetThread()->GetTHAllocContextObj();
+
+    //    if (th != 0)
+    //    {
+    //        th.GetName(strTypeName);
+    //        name = strTypeName.GetUnicode();
+    //        typeId = th.GetMethodTable();
+    //    }
+    //}
+    //EX_CATCH{}
+    //EX_END_CATCH(SwallowAllExceptions)
+    // end of duplication
+
+    if (typeId != nullptr)
+    {
+        unsigned int allocKind =
+            (flags & GC_ALLOC_PINNED_OBJECT_HEAP) ? 2 :
+            (flags & GC_ALLOC_LARGE_OBJECT_HEAP) ? 1 :
+            0;  // SOH
+        unsigned int heapIndex = 0;
+#ifdef BACKGROUND_GC
+        gc_heap* hp = gc_heap::heap_of((BYTE*)orObject);
+        heapIndex = hp->heap_number;
+#endif
+        FireEtwAllocationSampled(allocKind, GetClrInstanceId(), typeId, name, heapIndex, (BYTE*)orObject, size, samplingBudgetOffset);
```

Comment:
You may already have this on a list somewhere, but `name` is always `null` (I'm not sure that nativeaot even has the type name data available...)

Reply:
> You may already have this on a list somewhere, but `name` is always `null` (I'm not sure that nativeaot even has the type name data available...)

I'm reusing the same code as AllocationTick to get the type name and it is working in the tests for Core. I'll try the same code for NativeAOT when the compilations errors are fixed.

Reply:
Sounds good.  I suspect that nativeaot doesn't even have the type names, so it may just always need to be null, but wanted to call it out.

Reply:
Sorry @brianrob!
I missed that your point was for NativeAOT only!